cli/src/merge_tools/builtin.rs

```diff
@@ -424,8 +425,133 @@ pub fn edit_diff_builtin(
     Ok(tree_id)
 }
 
+fn make_merge_sections(
+    merge_result: MergeResult,
+    file_merge: Merge<Option<FileId>>,
+) -> Result<Vec<scm_record::Section<'static>>, BuiltinToolError> {
+    let mut sections = Vec::new();
+    match merge_result {
+        MergeResult::Resolved(ContentHunk(buf)) => {
+            let right_id = file_merge.adds().get(1).and_then(|side| side.clone());
+            let contents = buf_to_file_contents(right_id.map(|id| id.hex()), buf);
+            let section = match contents {
+                FileContents::Absent => None,
+                FileContents::Text {
+                    contents,
+                    hash: _,
+                    num_bytes: _,
+                } => Some(scm_record::Section::Unchanged {
+                    lines: contents
+                        .split_inclusive('\n')
+                        .map(|line| Cow::Owned(line.to_owned()))
+                        .collect(),
+                }),
+                FileContents::Binary { hash, num_bytes } => Some(scm_record::Section::Binary {
+                    is_checked: false,
+                    old_description: None,
+                    new_description: Some(Cow::Owned(describe_binary(hash.as_deref(), num_bytes))),
+                }),
+            };
+            if let Some(section) = section {
+                sections.push(section);
+            }
+        }
+        MergeResult::Conflict(hunks) => {
+            for hunk in hunks {
+                let (removes, adds) = hunk.take();
+                sections.push(match adds.as_slice() {
+                    [ContentHunk(contents)] => {
+                        let contents = std::str::from_utf8(contents).map_err(|err| {
+                            BuiltinToolError::DecodeUtf8 {
+                                source: err,
+                                item: "unchanged hunk",
+                            }
+                        })?;
+                        scm_record::Section::Unchanged {
+                            lines: contents
+                                .split_inclusive('\n')
+                                .map(|line| Cow::Owned(line.to_owned()))
+                                .collect(),
+                        }
+                    }
+                    _ => {
+                        let lines: Vec<scm_record::SectionChangedLine> = adds
+                            .into_iter()
+                            .interleave(removes)
```

Comment:
If you used `into_resolved()` above, you'll still have a `hunk` here, and you can use `hunk.into_iter()` to get the interleaved order.